lara-typescript/src/interactive-api-client/api.ts

```diff
@@ -388,3 +391,67 @@ export const getInteractiveSnapshot = (options: IGetInteractiveSnapshotOptions)
 export const getLibraryInteractiveList = (options: IGetLibraryInteractiveListRequest) => {
   THROW_NOT_IMPLEMENTED_YET("getLibraryInteractiveList");
 };
+
+export const writeAttachment = (params: Omit<IWriteAttachmentRequest, "requestId" | "operation">) => {
+  return new Promise<void>((resolve, reject) => {
+    const client = getClient();
+    const { name, content, options } = params;
+    const request: IAttachmentUrlRequest = { name, operation: "write", requestId: client.getNextRequestId() };
+    client.addListener("attachmentUrl", async (response: IAttachmentUrlResponse) => {
+      if (response.url) {
+        try {
+          await fetch(response.url, { ...options, method: "PUT", body: content });
+          resolve();
+        }
+        catch (e) {
+          reject(e);
+        }
+      }
+      else {
+        reject(new Error(response.error || "error writing attachment"));
+      }
+    }, request.requestId);
+    client.post("getAttachmentUrl", request);
+  });
+};
+
+export const readAttachment = (name: string) => {
+  return new Promise<Response>((resolve, reject) => {
+    // set up response listener
+    const client = getClient();
+    const request: IAttachmentUrlRequest = { name, operation: "read", requestId: client.getNextRequestId() };
+    client.addListener("attachmentUrl", async (response: IAttachmentUrlResponse) => {
+      if (response.url) {
+        try {
+          // resolves with the fetch Response object, so clients can check status
+          // or convert to JSON or text or something else.
+          resolve(await fetch(response.url));
+        }
+        catch (e) {
+          reject(e);
+        }
+      }
+      else {
+        reject(new Error(response.error || "error reading attachment"));
+      }
+    }, request.requestId);
+    client.post("getAttachmentUrl", request);
+  });
+};
+
+export const getAttachmentUrl = (name: string) => {
```

Comment:
Because these URLs will expire. It would be useful if an optional timeout (in seconds) was passed as an argument.  That way an interactive that wants a URL that it can reuse during a session could specify a few hours. While URL that is sensitive and going to be used immediately could have a short timeout of a few seconds.

Reply:
I was thinking that that should be a host decision (in the activity-player draft PR it defaults to five minutes for writes and two hours for reads) but I'm happy to make it client-configurable if you think that makes sense.